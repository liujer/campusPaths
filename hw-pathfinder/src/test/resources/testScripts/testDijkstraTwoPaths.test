# Find a path using Dijkstra's algorithm when there are two paths
# with identical weight
# Should add based on when node was first added
CreateGraph A
AddNode A one
AddNode A two
AddNode A three
AddNode A four

AddEdge A one two 2
AddEdge A one three 2
AddEdge A two four 1
AddEdge A three four 1

FindPath A one four