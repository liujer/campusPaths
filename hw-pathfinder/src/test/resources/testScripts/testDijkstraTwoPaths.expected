# Find a path using Dijkstra's algorithm when there are two paths
# with identical weight
# Should add based on when node was first added
created graph A
added node one to A
added node two to A
added node three to A
added node four to A

added edge 2.000 from one to two in A
added edge 2.000 from one to three in A
added edge 1.000 from two to four in A
added edge 1.000 from three to four in A

path from one to four:
one to two with weight 2.000
two to four with weight 1.000
total cost: 3.000
